demos/platforms/stm32f103c8.cpp

```diff
@@ -55,6 +57,13 @@ void initialize_platform(resource_list& p_resources)
   static hal::stm32f1::input_pin input_pin('B', 4);
   p_resources.input_pin = &input_pin;
 
+  static hal::atomic_spin_lock my_lock;
```

Comment:
```suggestion
  static hal::atomic_spin_lock adc_lock;
```

Lets rename this to what it has been used for.